include/swift/Basic/SourceManager.h

```diff
@@ -94,6 +94,7 @@ class SourceManager {
 private:
   llvm::SourceMgr LLVMSourceMgr;
   llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> FileSystem;
+  bool editorMode = false;
```

Comment:
suggest: IDEMode (more consistent with the below)

Reply:
I picked the name `editorMode` based on the flag, and that it seems more fitting in terms of what's trying to be expressed (that is, we should disable memory mapping when a source file is open in any editor, even if that editor would not typically classify as an IDE).